dbt_meshify/dbt_meshify.py

```diff
@@ -33,10 +35,12 @@ def add_model_contract_to_yml(self, full_yml_dict: Dict[str, str], model_catalog
         # update the model entry in the full yml file
         # if no entries exist, add the model entry
         # otherwise, update the existing model entry in place
+        model_ordered_dict.update(model_yml)
+        model_ordered_dict = {k: v for k, v in model_ordered_dict.items() if v is not None}
         if not full_yml_dict.get("models"):
-            full_yml_dict.update({"models": [model_yml]})
+            full_yml_dict.update({"models": [model_ordered_dict]})
         else:
             for i, entry in enumerate(full_yml_dict.get("models", [])):
                 if entry["name"] == model_name:
-                    full_yml_dict["models"][i] = model_yml
+                    full_yml_dict["models"][i] = model_ordered_dict
```

Comment:
I suspect we'll eventually want to manage an internal dictionary structure for managing models. This way, we will be able to pass in the name of the model and directly update the values without needing to iterate through a list.

Reply:
agreed here -- what might that look like roughly speaking?

Reply:
One solution would be to have a variable that stores the dictionary, and to grab the dictionary items for the models list just before returning. Maybe something like:

```python

models = { model['name']: model for model in full_yml_dict['models'] }

# ... stuff already in the method ...

# Update the model entry
model_name = model_ordered_dict['name']
models[model_name] = model_ordered_dict

# ... anything else needed ...

# return the yml
full_yml_dict['models'] = list(models.values())
return full_yml_dict
```

Reply:
made this change as well -- think i could see the argument for us maintaining a ModelYml class that handles the allowed keys, data types, and order of keys we want to display, so it could endup like 

```python
models = { model['name']: ModelYml(model) for model in full_yml_dict['models'] }
```